src/components/PasteButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { GestureResponderEvent, StyleSheet, Text } from 'react-native'
+import Touchable from 'src/components/Touchable'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { useClipboard } from 'src/utils/useClipboard'
+
+interface Props {
+  value: string
+  setValue: React.Dispatch<React.SetStateAction<string>>
+  onPress?: (event: GestureResponderEvent) => void
+}
+
+export const PasteButton = ({ value, setValue, onPress }: Props) => {
+  const { t } = useTranslation()
+  const [, , getFreshClipboardContent] = useClipboard()
+
+  const pasteClipboard = async (event: GestureResponderEvent) => {
+    const content = await getFreshClipboardContent()
+    setValue(content)
+    onPress?.(event)
+  }
+
+  // when a value is already set, remove the button
+  if (value) return null
```

Comment:
this also feels like this component needs to know too much about the consumer. i think it makes more sense if 
1. the component takes a prop for `isVisible`
2. the consumer should show or hide the paste button

as a general rule of thumb, generic components are more easily reused if the consumer has control of them. avoiding as many conditions as possible, and making props as instructive as possible usually helps with this. (e.g. having a `value` prop that actually sometimes hides the component is not obvious, and may not work for other scenarios. but having something like `isVisible` allows the consumer to decide the logic.)

having said this, i actually prefer option 2. because there's less props to manage.

Reply:
I chose Option 2.